src/dmd/astbase.d

```diff
@@ -1331,13 +1335,13 @@ struct ASTBase
         extern (C++) __gshared AggregateDeclaration moduleinfo;
 
         const FileName srcfile;
-        const(char)* arg;
+        const(char)[] arg;
 
-        extern (D) this(const(char)* filename, Identifier ident, int doDocComment, int doHdrGen)
+        extern (D) this(const ref Loc loc, const(char)[] filename, Identifier ident, int doDocComment, int doHdrGen)
         {
-            super(ident);
+            super(loc, ident);
             this.arg = filename;
-            srcfile = FileName(FileName.defaultExt(filename.toDString, mars_ext));
+            srcfile = FileName(filename);
         }
 
```

Comment:
Keep the old signature.

```suggestion

extern (D) this(const(char)* filename, Identifier ident, int doDocComment, int doHdrGen)
{
    this(Loc.initial, filename.toDString, ident, doDocComment, doHdrGen);
}
```

Reply:
ping @dorianverna17

Reply:
I also kept the old signature in the commit made yesterday